src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStoreV1.swift

```diff
@@ -0,0 +1,749 @@
+			"""
+
+			guard database.executeStatements(sql) else {
+				Log.error("[ContactDiaryStore] (\(database.lastErrorCode())) \(database.lastErrorMessage())", log: .localData)
+				result = .failure(SQLiteErrorCode(rawValue: database.lastErrorCode()) ?? SQLiteErrorCode.unknown)
+				return
+			}
+
+			let updateDiaryDaysResult = updateDiaryDays(with: database)
+			guard case .success = updateDiaryDaysResult else {
+				Log.error("[ContactDiaryStore] (\(database.lastErrorCode())) \(database.lastErrorMessage())", log: .localData)
+				result = .failure(SQLiteErrorCode(rawValue: database.lastErrorCode()) ?? SQLiteErrorCode.unknown)
+				return
+			}
+
+			result = .success(())
+		}
+
+		guard let _result = result else {
+			fatalError("Result should not be nil.")
+		}
+
+		return _result
+	}
+	
+	// MARK: - Private
+
+	private let dataRetentionPeriodInDays = 16
+	private let key: String
+
+	private var dateFormatter: ISO8601DateFormatter = {
+		let dateFormatter = ISO8601DateFormatter()
+		dateFormatter.formatOptions = [.withFullDate]
+		return dateFormatter
+	}()
+
+
+	private let databaseQueue: FMDatabaseQueue
+
+	private func createSchemaIfNeeded(schema: ContactDiaryStoreSchemaV1) {
+		_ = schema.create()
+	}
+
+	private func openAndSetup() {
+		databaseQueue.inDatabase { database in
+			Log.info("[ContactDiaryStore] Open and setup database.", log: .localData)
+
+			let dbhandle = OpaquePointer(database.sqliteHandle)
+			guard CWASQLite.sqlite3_key(dbhandle, key, Int32(key.count)) == SQLITE_OK else {
+				Log.error("[ContactDiaryStore] Unable to set Key for encryption.", log: .localData)
+				return
+			}
+
+			guard database.open() else {
+				Log.error("[ContactDiaryStore] Database could not be opened", log: .localData)
+				return
+			}
+
+			let sql = """
+				PRAGMA locking_mode=EXCLUSIVE;
+				PRAGMA auto_vacuum=2;
+				PRAGMA journal_mode=WAL;
+				PRAGMA foreign_keys=ON;
+			"""
+			guard database.executeStatements(sql) else {
+				Log.error("[ContactDiaryStore] (\(database.lastErrorCode())) \(database.lastErrorMessage())", log: .localData)
+				return
+			}
+		}
+	}
+
+	private func registerToDidFinishLaunchingNotification() {
+		NotificationCenter.default.addObserver(self, selector: #selector(didBecomeActiveNotification), name: UIApplication.didBecomeActiveNotification, object: nil)
+	}
+
+	@objc
+	private func didBecomeActiveNotification(_ notification: Notification) {
+		_ = cleanup()
+	}
+
+	private func fetchContactPersons(for date: String, in database: FMDatabase) -> Result<[DiaryContactPerson], SQLiteErrorCode> {
+		var contactPersons = [DiaryContactPerson]()
+
+		let sql = """
+				SELECT ContactPerson.id AS contactPersonId, ContactPerson.name, ContactPersonEncounter.id AS contactPersonEncounterId
+				FROM ContactPerson
+				LEFT JOIN ContactPersonEncounter
+				ON ContactPersonEncounter.contactPersonId = ContactPerson.id
+				AND ContactPersonEncounter.date = ?
+				ORDER BY ContactPerson.name ASC, contactPersonId ASC
```

Comment:
This is case sensitive, maybe better sort it case insensitive?
```suggestion
				ORDER BY ContactPerson.name COLLATE NOCASE ASC, contactPersonId COLLATE NOCASE ASC
```

Reply:
Good point. Will change it.